addons/event_booth_sale/static/src/js/event_booth_configurator_widget.js

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * Extension of the ProductConfiguratorWidget to support event booth configuration.
+ * It opens when an event booth product_product is set.
+ *
+ * The event booth information include:
+ * - event_id
+ * - event_booth_id
+ *
+ */
+ProductConfiguratorWidget.include({
+
+    //--------------------------------------------------------------------------
+    // Overrides
+    //--------------------------------------------------------------------------
+
+    /**
+     * @returns {boolean}
+     *
+     * @override
+     * @private
+     */
+    _isConfigurableLine: function () {
+        return this.recordData.is_event_booth || this._super.apply(this, arguments);
+    },
+
+    /**
+     * Opens the event booth configurator in 'edit' mode
+     *
+     * @override
+     * @private
+     */
+    _onEditLineConfiguration: function () {
+        if (this.recordData.is_event_booth) {
+            let defaultValues = {
+                default_product_id: this.recordData.product_id.data.id,
+                default_sale_order_line_id: this.recordData.id || null
+            };
+            if (this.recordData.event_id) {
+                defaultValues.default_event_id = this.recordData.event_id.data.id;
+            }
+            if (this.recordData.event_booth_category_id) {
+                defaultValues.default_event_booth_category_id = this.recordData.event_booth_category_id.data.id;
+            }
+            if (this.recordData.event_booth_pending_ids) {
+                defaultValues.default_event_booth_ids = this.recordData.event_booth_pending_ids.res_ids;
+            }
+            this._openEventBoothConfigurator(defaultValues, this.dataPointID);
+        } else {
+            this._super.apply(this, arguments);
+        }
+    },
+
+    /**
+     * @param {integer} productId: product.product ID
+     * @param {String} dataPointId
+     * @returns {Promise<Boolean>} stopPropagation true if a suitable configurator has been found.
+     *
+     * @override
+     * @private
+     */
+    _onProductChange: function (productId, dataPointId) {
+        return this._super.apply(this, arguments).then((stopPropagation) => {
+            if (stopPropagation) {
+                return Promise.resolve(true);
+            } else {
+                return this._isProductEventBooth(productId, dataPointId);
+            }
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //-------------------------------------------------------------------------
+
+    /**
+     * This method will check if the productId needs a configuration or not.
+     *
+     * @param {integer} productId: product.product ID
+     * @param {String} dataPointId
+     * @returns {Promise<Boolean>} stopPropagation true if the product is an event booth.
+     *
+     * @private
+     */
+    _isProductEventBooth: function (productId, dataPointId) {
```

Comment:
We will somehow have to refactor those some day.
Currently if you install this module, event and rental, it triggers 3 different read calls on the product.product model.